JoomGalleryfriends/JG4-dev
# Reset language file (Backend)

### New Naming logic
This pull request resets the language file for the backend and adds a new logic to the naming of the constants. The constants are now named based on its content rather than the location where its used.

Example JG3:
COM_JOOMGALLERY_CATMAN_HIDDEN="Hidden"

Example JG4:
COM_JOOMGALLERY_HIDDEN="Hidden"

### Usage of Core strings
Additionally the PR tries to use language constants from Joomla core whenever possible. This allows us to reduce the language strings to be maintained a lot.

Example JG3:
COM_JOOMGALLERY_COMMON_CATEGORY="Category"

Example JG4:
JCATEGORY="Category"

### Test instructions
You have to deinstall your currently installed JommGallery 4 version and reinstall it using this code. Otherwise the language files will mess up with previous installed versions!

- Switch through the backend GUI and look for untranslated strings. Report untranslated strings hre in the comment section of the PR
- Read through the english texts provided in the backend GUI. Report if you see any texts that does not make sense or are not good quality. Always provide your suggestion for your reported text.
- Untranslated strings in Config Manager>>General Settings>>Uploads>>Use image metadata>>Source Metadata Field will be threated in a different PR
